### DISCUSSION TITLE
How do I get all the layers in forecast/hourly?

### DISCUSSION THREAD

**Discussion by mcao1224** [weather-gov/api#260] [Q&A]
I am looking for hourly forecast weather data. By applying https://api.weather.gov/gridpoints/LWX/96,68/forecast/hourly, I can only get limited weather attributes like temperature, wind speed, wind direction, shortForecast. How can I get the other attributes like relative humidity, dew point, cloud coverage, etc? Thank you!

**Comment by jfbauer432** [comment]
There is a lot more attributes at https://api.weather.gov/gridpoints/LWX/96,68

**Comment by mcao1224** [comment]
Thanks @jfbauer432 !

**Comment by ddownn** [comment]
I have been trying to parse data from the url such as 
https://api.weather.gov/gridpoints/LWX/96,68

But I have found that my translations break when I switch to another grid point that returns additional elements that my definition doesn't have. This is because I based my definitions on a sample result, and I cannot find a complete Json schema for the values that could be returned for this type of url. 

What could be the solution for this problem?

**Comment by scadergit** [comment]
In general, if would be more future proof for the application to ignore additional parameters until needed. The gridpoint is a dump of the forecast data provided by the weather offices and could change (hopefully with a SCN). This might also be helpful: https://github.com/weather-gov/api/issues/144

**Comment by ddownn** [comment]
Yes, that is helpful thank you. 

It could be that I'm just not doing something right since this is my first time trying to parse JSON. I thought everything was working until I switched to a completely different gridpoint. 

It seems that my main problem is that the automated generator I used defined "values" as an integer data type for some elements due to the sample data I used, but it neglected to identify that when "values" is encountered, it is always an array
```
"properties" : {
   "category": {
      "uom": "wmoUnit:mm",
      "values": []
   },
}
```
...and so decoding broke when the data from another location had an array of values instead of a single integer. 

Thanks for your reply
